### PR TITLE
pldm: Adding Fix for NVRAM checksum corruption

### PR DESCRIPTION
**Pull request by Patel-Kamalkumar**
When user do chasisoff then power is abruptly down for chassis and NVRAM file transfer is abruptly stops which leads to mismatch NVRAM file checksum and PHYP updated checksum due to data corruption. This commit adds the synchronous file transfer mechanism only for NVRAM and Checksum file to avoid race condition while chassis off.

### CODE REVIEW

File: oem/ibm/libpldmresponder/utils.hpp [hunk 1]
```diff
@@ -29,13 +28,11 @@ struct CustomFD
     CustomFD(CustomFD&&) = delete;
     CustomFD& operator=(CustomFD&&) = delete;
 
-    CustomFD(int fd, bool closeOnOutScope = true) :
-        fd(fd), closeOnOutScope(closeOnOutScope)
-    {}
+    CustomFD(int fd) : fd(fd) {}
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/utils.hpp:31]: what happens to the fd in async case? do we not need the outOfScope check?

File: oem/ibm/libpldmresponder/file_io.hpp [hunk 2]
```diff
@@ -122,13 +122,23 @@ class DMA
...
      */
-    int getNewXdmaFd()
+    int getNewXdmaFd(bool isBlckSocket = true)
     {
         try
         {
-            xdmaFd = open(xdmaDev, O_RDWR | O_NONBLOCK);
+            if (isBlckSocket)
+            {
+                xdmaFd = open(xdmaDev, O_RDWR | O_NONBLOCK);
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.hpp:136]: here you are checking if its a blockSocket and then opening it in O_NONBLOCK mode.

**Review by Pavithrab7 — CHANGES_REQUESTED** [review]

File: oem/ibm/libpldmresponder/file_io.hpp (hunk 2 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/file_io.hpp:136]: yea correct, will change

File: oem/ibm/libpldmresponder/utils.hpp (hunk 1 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/utils.hpp:31]: this is the extra CustomFD class in async case which was not used but in sync case it's required to close when object out of scope.
in async case, DMA class handles both file descriptor

**Review by manojkiraneda — APPROVED** [review]

**Review by jaypadath — APPROVED** [review]

**Review by Pavithrab7 — APPROVED** [review]

**Pull request merged**